openmetadata-ui/src/main/resources/ui/src/pages/DataModelPage/DataModelPage.component.tsx

```diff
@@ -374,6 +413,86 @@ const DataModelsPage = () => {
     }
   };
 
+  // Lineage handlers
+  const handleAddLineage = async (edge: Edge) => {
+    try {
+      await addLineage(edge);
+    } catch (error) {
+      showErrorToast(error as AxiosError);
+    }
+  };
+
+  const handleRemoveLineage = async (data: EdgeData) => {
+    try {
+      await deleteLineageEdge(
+        data.fromEntity,
+        data.fromId,
+        data.toEntity,
+        data.toId
+      );
+    } catch (error) {
+      showErrorToast(error as AxiosError);
+    }
+  };
+
+  const handleSetLeafNode = (val: EntityLineage, pos: LineagePos) => {
+    if (pos === 'to' && val.downstreamEdges?.length === 0) {
+      setLeafNodes((prev) => ({
+        ...prev,
+        downStreamNode: [...(prev.downStreamNode ?? []), val.entity.id],
+      }));
+    }
+    if (pos === 'from' && val.upstreamEdges?.length === 0) {
+      setLeafNodes((prev) => ({
+        ...prev,
+        upStreamNode: [...(prev.upStreamNode ?? []), val.entity.id],
+      }));
+    }
+  };
+
+  const handleLoadLineageNode = async (
+    node: EntityReference,
+    pos: LineagePos
+  ) => {
+    setNodeLoading({ id: node.id, state: true });
+
+    try {
+      const response = await getLineageByFQN(
+        node.fullyQualifiedName ?? '',
+        node.type
+      );
+      handleSetLeafNode(response, pos);
+      setEntityLineage(getEntityLineage(entityLineage, response, pos));
+      setTimeout(() => {
+        setNodeLoading((prev) => ({ ...prev, state: false }));
+      }, 500);
+    } catch (error) {
+      showErrorToast(error as AxiosError);
+    }
+  };
+
+  const handleFullScreenClick = () =>
+    history.push(
+      getLineageViewPath(EntityType.DASHBOARD_DATA_MODEL, dashboardDataModelFQN)
+    );
+
+  const fetchTabSpecificData = () => {
+    switch (tab) {
+      case DATA_MODELS_DETAILS_TABS.LINEAGE: {
+        fetchLineageData(dashboardDataModelFQN);
+
+        break;
+      }
+
+      default:
+        break;
+    }
+  };
+
+  useEffect(() => {
+    fetchTabSpecificData();
+  }, [tab, dashboardDataModelFQN]);
+
```

Comment:
We can reduce it to one

Reply:
@chirag-madlani  updated it!

Reply:
@Ashish8689 point over here was we don't need that switch case

Reply:
@chirag-madlani  Actually we need it as we are fetching lineage, then activity too based on the tab. So in that case we require switch.

Reply:
@Ashish8689 Lineage should be fetched within Lineage component itself

Reply:
@chirag-madlani  updated as per comments!